VersionControlUI/src/DiffManager.cpp

```diff
@@ -239,11 +261,37 @@ void DiffManager::createOverlaysForChanges(Visualization::ViewItem* diffViewItem
 
 		for (auto item : items)
 		{
+			allItemsToScale.insert(item);
 			auto overlay = new Visualization::HighlightOverlay{item,
 					Visualization::HighlightOverlay::itemStyles().get(highlightOverlayStyle)};
 			item->addOverlay(overlay, highlightOverlayName);
 		}
 	}
+
+	QSet<Visualization::Item*> removeItems = findAllItemsWithParentPresent(allItemsToScale);
+
+	auto it = allItemsToScale.begin();
```

Comment:
Know your libraries, instead of these several lines, use simply:
http://doc.qt.io/qt-5/qset.html#subtract